### PR TITLE
[feature/redux] Add user, query, params, ip to runtime state

### PR DESCRIPTION
**Pull request by alexsuslov**

### CODE REVIEW

File: src/server.js [hunk 1]
```diff
@@ -108,11 +108,19 @@ app.get('*', async (req, res, next) => {
       cookie: req.headers.cookie,
       history,
     });
+    // user, query, params
+    ['user', 'query', 'params'].map(v => (
```
Comment (langpavel) [inline src/server.js]: I believe this should be all handled differently.

- `user` should have it's own action and reducer
- `query` and `params` should be handled wit some `history` to `redux` binding similar to [redux-router](`https://github.com/acdlite/redux-router`)

File: src/server.js [hunk 2]
```diff
@@ -108,11 +108,19 @@ app.get('*', async (req, res, next) => {
...
+      store.dispatch(setRuntimeVariable({ name: v, value: req[v] }))
+    ));
 
     store.dispatch(setRuntimeVariable({
       name: 'initialNow',
       value: Date.now(),
     }));
+    // ip
+    const ip = req.headers['x-forwarded-for'] || req.connection.remoteAddress;
+    store.dispatch(setRuntimeVariable({ name: 'ip', value: ip }));
```
Comment (langpavel) [inline src/server.js]: `setRuntimeVariable` is there mostly for example purposes.. but this is one of good use cases :+1:

**Comment by alexsuslov** [comment]
HI!
As you please.
-- 
Alex Suslov
Sent with Airmail

От 13 октября 2016 г. в 3:23:31, Pavel Lang (notifications@github.com)
написал:

*@langpavel* commented on this pull request.

I will leave this as example but will probably not merge this changes
Anyway Thank you!
------------------------------

In src/server.js
<https://github.com/kriasoft/react-starter-kit/pull/854#pullrequestreview-3986879>
:

> @@ -108,11 +108,19 @@ app.get('*', async (req, res, next) => {
       cookie: req.headers.cookie,
       history,
     });
+    // user, query, params
+    ['user', 'query', 'params'].map(v => (

I believe this should be all handled differently.

   - user should have it's own action and reducer
   - query and params should be handled wit some history to redux binding
   similar to redux-router

------------------------------

In src/server.js
<https://github.com/kriasoft/react-starter-kit/pull/854#pullrequestreview-3986879>
:

>
     store.dispatch(setRuntimeVariable({
       name: 'initialNow',
       value: Date.now(),
     }));
+    // ip
+    const ip = req.headers['x-forwarded-for'] || req.connection.remoteAddress;
+    store.dispatch(setRuntimeVariable({ name: 'ip', value: ip }));

setRuntimeVariable is there mostly for example purposes.. but this is one
of good use cases 👍

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/kriasoft/react-starter-kit/pull/854#pullrequestreview-3986879>,
or mute the thread

**Comment by alexsuslov** [comment]
I sent google auth to your branch?

-- 
Alex Suslov
Sent with Airmail

От 13 октября 2016 г. в 3:23:31, Pavel Lang (notifications@github.com)
написал:

*@langpavel* commented on this pull request.

I will leave this as example but will probably not merge this changes
Anyway Thank you!
------------------------------

In src/server.js
<https://github.com/kriasoft/react-starter-kit/pull/854#pullrequestreview-3986879>
:

> @@ -108,11 +108,19 @@ app.get('*', async (req, res, next) => {
       cookie: req.headers.cookie,
       history,
     });
+    // user, query, params
+    ['user', 'query', 'params'].map(v => (

I believe this should be all handled differently.

   - user should have it's own action and reducer
   - query and params should be handled wit some history to redux binding
   similar to redux-router

------------------------------

In src/server.js
<https://github.com/kriasoft/react-starter-kit/pull/854#pullrequestreview-3986879>
:

>
     store.dispatch(setRuntimeVariable({
       name: 'initialNow',
       value: Date.now(),
     }));
+    // ip
+    const ip = req.headers['x-forwarded-for'] || req.connection.remoteAddress;
+    store.dispatch(setRuntimeVariable({ name: 'ip', value: ip }));

setRuntimeVariable is there mostly for example purposes.. but this is one
of good use cases 👍

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/kriasoft/react-starter-kit/pull/854#pullrequestreview-3986879>,
or mute the thread

**Comment by ulani** [comment]
@alexsuslov  thank you very much for this PR! Unfortunately, we have close it due to inactivity. Feel free to re-open it or [join](https://discord.com/invite/2nKEnKq) our [Discord](https://discord.com/invite/2nKEnKq) channel for [discussion](https://github.com/kriasoft/react-starter-kit/discussions/1950).

NOTE: The `main` branch has been updated with React Starter Kit v2, using JAM-style architecture.

**Pull request closed without merging**